api/src/main/java/brooklyn/basic/AbstractBrooklynObjectSpec.java

```diff
@@ -121,4 +121,21 @@ protected final void checkIsImplementation(Class<?> val, Class<? super T> requir
         if (Modifier.isAbstract(val.getModifiers())) throw new IllegalStateException("Implementation "+val+" is abstract, but must be a non-abstract class");
     }
 
+    @Override
+    public boolean equals(Object obj) {
+        if (obj==null) return false;
```

Comment:
I'm slightly hesitant about equals/hashCode on mutable objects. It potentially breaks the contract if two objects that were equal then become not-equal and the hashCode changes (http://stackoverflow.com/a/4718059/1393883).

Guessing this is just used for tests? Perhaps having a utility class for comparisons would be safer?